openpathsampling/tests/test_data/ops1_api.txt

```diff
@@ -0,0 +1,270 @@
+openpathsampling.PrefixTrajectoryEnsemble
+openpathsampling.PyEMMAFeaturizerCV
+openpathsampling.RandomAllowedChoiceMover
+openpathsampling.RandomChoiceMoveChange
+openpathsampling.RandomChoiceMover
+openpathsampling.RandomSubtrajectorySelectMover
+openpathsampling.RandomVelocities
+openpathsampling.RejectedMaxLengthSampleMoveChange
+openpathsampling.RejectedNaNSampleMoveChange
+openpathsampling.RejectedSampleMoveChange
+openpathsampling.RelativeComplementVolume
+openpathsampling.ReplicaExchangeMover
+openpathsampling.ReplicaNetwork
+openpathsampling.ReplicaNetworkGraph
+openpathsampling.ReversedTrajectoryEnsemble
+openpathsampling.SRTISBiasFromNetwork
+openpathsampling.SRTISScheme
+openpathsampling.Sample
+openpathsampling.SampleDetails
+openpathsampling.SampleMoveChange
+openpathsampling.SampleMover
+openpathsampling.SampleSet
+openpathsampling.SelectionMover
+openpathsampling.SequentialEnsemble
+openpathsampling.SequentialMoveChange
+openpathsampling.SequentialMover
+openpathsampling.ShootFromSnapshotsSimulation
+openpathsampling.ShootingPointAnalysis
+openpathsampling.ShootingPointSelector
+openpathsampling.SingleAtomVelocityDirectionModifier
+openpathsampling.SingleFrameEnsemble
+openpathsampling.SingleReplicaMinusMover
+openpathsampling.SnapshotByCoordinateDict
+openpathsampling.StateSwapMover
+openpathsampling.StepVisualizer2D
+openpathsampling.Storage
+openpathsampling.SubMoveChange
+openpathsampling.SubPathMover
+openpathsampling.SubtrajectorySelectMover
+openpathsampling.SuffixTrajectoryEnsemble
+openpathsampling.SymmetricDifferenceVolume
+openpathsampling.TISEnsemble
+openpathsampling.TISTransition
+openpathsampling.TPSNetwork
+openpathsampling.TPSTransition
+openpathsampling.engines.Trajectory
+openpathsampling.TrajectorySegmentContainer
+openpathsampling.TrajectoryTransitionAnalysis
+openpathsampling.Transition
+openpathsampling.TransitionNetwork
+openpathsampling.TwoWayShootingMover
+openpathsampling.UniformSelector
+openpathsampling.UnionEnsemble
+openpathsampling.UnionVolume
+openpathsampling.VelocityDirectionModifier
+openpathsampling.Volume
+openpathsampling.VolumeCombination
+openpathsampling.VolumeEnsemble
+openpathsampling.VolumeInterfaceSet
+openpathsampling.WrappedEnsemble
+openpathsampling.condense_repeats
+openpathsampling.git_HEAD
+openpathsampling.join_ensembles
+openpathsampling.join_volumes
+openpathsampling.resources_directory
+openpathsampling.trace_ensembles_for_replica
+openpathsampling.trace_replicas_for_ensemble
+openpathsampling.analysis.PathHistogram
+openpathsampling.netcdfplus.StorableNamedObject
+openpathsampling.numerics.SparseHistogram
+openpathsampling.analysis.tis.ConditionalTransitionProbability
+openpathsampling.analysis.tis.DictFlux
+openpathsampling.analysis.tis.EnsembleHistogrammer
+openpathsampling.analysis.tis.FullHistogramMaxLambdas
+openpathsampling.analysis.tis.MinusMoveFlux
+openpathsampling.analysis.tis.MultiEnsembleSamplingAnalyzer
+openpathsampling.analysis.tis.PathLengthHistogrammer
```

Comment:
is this only implemented for tis and not for TPS?

Reply:
The specific `analysis.tis.PathLengthHistogrammer` should work on either TPS or TIS data, but its benefit is that it returns path length histograms per ensemble (as needed for TIS). As with other per-ensemble computations in the `analysis.tis` namespace, this first goes through all the steps and finds the (Monte Carlo) weight of each trajectory within each ensemble. Then, in a second pass, it calculates the value (length in this case) and updates the histogram with the correct weight.

When there are multiple ensembles and/or an expensive value to calculate, this approach is efficient. But for TPS, there's only one ensemble, and the length is cheap to calculate. So the path length histogram can be plotted easily (using only one pass over the data) with:

```python
plt.hist([len(s.active[0].trajectory) for s in steps])
```